commons-test/src/main/java/org/infinispan/commons/test/PolarionTestNameListener.java

```diff
@@ -0,0 +1,70 @@
+package org.infinispan.commons.test;
+
+import java.lang.reflect.Field;
+
+import org.testng.ITestContext;
+import org.testng.ITestListener;
+import org.testng.ITestResult;
+import org.testng.internal.TestResult;
+
+/**
+ *
+ * This listener fixes the method name by overwriting the private `TestResult.m_name` field and performs some
+ * additional processing to ensure the resulting test name is unique.
```

Comment:
The javadoc you posted in the chat had 2 more paragraphs

Reply:
The class was renamed to `TestNGTestNameListener`.  The javadoc was moved to `TestNGNameTranslator` that where is the hack is present.